File: 0023-merge-k-sorted-lists/0023-merge-k-sorted-lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeKLists(lists []*ListNode) *ListNode {
    
    if len(lists) == 0 {
        return nil
    }
    
    arr := []int{}
    
    for i := 0; i < len(lists); i++ {
        tmpList := lists[i]
        for tmpList != nil {
            arr = append(arr, tmpList.Val)
            tmpList = tmpList.Next
        }
    }
    
    if len(arr) == 0 {
        return nil
    }
    
    sort.Ints(arr)
    
    dummy := &ListNode{}
    cur := dummy
    
    for _, val := range arr {
        cur.Next = &ListNode{Val: val}
        cur = cur.Next
    }
    
    return dummy.Next
}

File: 0023-merge-k-sorted-lists/0023-merge-k-sorted-lists.php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution {

    /**
     * @param ListNode[] $lists
     * @return ListNode
     */
    function mergeKLists($lists) {
        $arr = [];
        foreach($lists as $list){
            while($list){
                $arr[] = $list->val;
                $list = $list->next;
            }
        }
        
        sort($arr);
        $dummy = new ListNode();
        $curr = $dummy;
        
        foreach($arr as $val){
            $curr->next = new ListNode($val);
            $curr = $curr->next;
        }
        
        return $dummy->next;
    }
}